src/Entity/FieldableEdgeEntityBase.php

```diff
@@ -0,0 +1,544 @@
+   *
+   * @param string $name
+   *   Name of the base field.
+   * @param string $type
+   *   Type of the property.
+   *
+   * @return \Drupal\Core\Field\BaseFieldDefinition|null
+   *   Base field definition if found, null otherwise.
+   */
+  protected static function getBaseFieldDefinition(string $name, string $type): ?BaseFieldDefinition {
+    $label = ucwords(preg_replace('/([a-z])([A-Z])/', '$1 $2', $name));
+    $is_array = strpos($type, 'list_') === 0;
+
+    try {
+      $definition = BaseFieldDefinition::create($type);
+    }
+    catch (\Exception $ex) {
+      // Type not found.
+      return NULL;
+    }
+
+    $definition->setLabel(t($label));
+    $definition->setCardinality($is_array ? FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED : 1);
+    $definition->setTranslatable(FALSE);
+    $definition->setDisplayConfigurable('view', TRUE);
+    $definition->setDisplayConfigurable('form', TRUE);
+
+    return $definition;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function get($field_name) {
+    if (!isset($this->fields[$field_name])) {
+      $value = $this->getFieldValue($field_name);
+
+      // Here field name equals the property name.
+      if ($value !== NULL) {
+        // Fix field value of a timestamp property field.
+        if (static::propertyFieldType($field_name) === 'timestamp') {
+          if (is_array($value)) {
+            $value = array_map(function ($item) {
+              /** @var \DateTimeImmutable $item */
+              return $item->getTimestamp();
+            }, $value);
+          }
+          else {
+            /** @var \DateTimeImmutable $value */
+            $value = $value->getTimestamp();
+          }
+        }
+      }
+
+      // Based on \Drupal\Core\Entity\ContentEntityBase::getTranslatedField().
+      /** @var \Drupal\Core\Field\FieldTypePluginManagerInterface $manager */
+      $manager = \Drupal::service('plugin.manager.field.field_type');
+      $this->fields[$field_name] = $manager->createFieldItemList($this, $field_name, $value);
+    }
+
+    return $this->fields[$field_name];
+  }
+
+  /**
+   * Returns the field value from the current object.
+   *
+   * @param string $field_name
+   *   Machine name of a field.
+   *
+   * @return mixed|null
+   *   Value of a field from current object, or null if it does exits.
+   */
+  protected function getFieldValue(string $field_name) {
+    $getter = 'get' . ucfirst($field_name);
+    // We call the getters on the current object instead of the decorated one
+    // because they can return the correct information.
+    // Because the current object implements the interface of the decorated
+    // object there should be any getter on the decorated object that does not
+    // have a decorator in the current class (that potentially also calls to the
+    // decorated getter method under the hood.)
+    if (method_exists($this, $getter)) {
+      return call_user_func([$this, $getter]);
+    }
+
+    return NULL;
```

Comment:
`\Drupal\apigee_edge\Entity\FieldableEdgeEntityBase::getProperties()` searches for `get` and `is` but this only looks for `get`.

Reply:
First comment: 
This how decorator pattern is working :) Decorator object should implement the interface (public behavior) of the decorated object.
At this moment we are not leveraging this behavior but it could be really useful later, for example when we have to transform the returned data from the decorated object before we would return it. Also, we may need to use Drupal entity object in a context where the SDK expects an SDK interface, like in SDK methods calls.

Second comment:
I do not get your point, "but this only looks for get.". Can you rephrase it?

Reply:
Maybe it should be something like: 

```
  /**
   * {@inheritdoc}
   */
  protected function getFieldValue(string $field_name) {
    $getter = 'get' . ucfirst($field_name);
    $boolean_accessor = 'is' . ucfirst($field_name);

    // We call the getters on the current object instead of the decorated one
    // because they can return the correct information.
    // Because the current object implements the interface of the decorated
    // object there should be any getter on the decorated object that does not
    // have a decorator in the current class (that potentially also calls to the
    // decorated getter method under the hood.)
    if (method_exists($this, $getter)) {
      return call_user_func([$this, $getter]);
    }
    elseif (method_exists($this, $boolean_accessor)) {
      return call_user_func([$this, $boolean_accessor]);
    }

    return NULL;
  }
```

Reply:
Ahh, now I get it which method you are talking about. Good catch! That could be problematic in the MINT entities. MGMT entities do not really have issers.